qiskit_nature/problems/second_quantization/base_problem.py

```diff
@@ -19,29 +19,53 @@
 
 from qiskit.opflow import PauliSumOp, Z2Symmetries
 
+from qiskit_nature import QiskitNatureError
 from qiskit_nature.drivers.second_quantization import BaseDriver, QMolecule, WatsonHamiltonian
 from qiskit_nature.converters.second_quantization import QubitConverter
+from qiskit_nature.properties.second_quantization.driver_result import DriverResult
 from qiskit_nature.results import EigenstateResult
+from qiskit_nature.transformers import BaseTransformer as LegacyBaseTransformer
 from qiskit_nature.transformers.second_quantization import BaseTransformer
 
 
 class BaseProblem(ABC):
     """Base Problem"""
 
-    def __init__(self, driver: BaseDriver, transformers: Optional[List[BaseTransformer]] = None):
+    def __init__(
+        self,
+        driver: BaseDriver,
+        transformers: Optional[List[Union[LegacyBaseTransformer, BaseTransformer]]] = None,
+    ):
         """
 
         Args:
             driver: A driver encoding the molecule information.
             transformers: A list of transformations to be applied to the molecule.
+
+        Raises:
+            QiskitNatureError: if new and legacy transformer instances are mixed.
         """
 
         self.driver = driver
         self.transformers = transformers or []
 
+        self._legacy_transform = False
+        new_transformers = False
+        for trafo in self.transformers:
+            if trafo is None:
+                continue
```

Comment:
Why is this needed - it supposed be a List[Transformer] not a List[Optional[Transformer]]?

Reply:
You're right! I reshuffled the code before and this should have been a check whether `transformers is None` because is actually is `Optional[List[Transformer]]`. But since I now set `self.transformers` first which will then be an empty list, this is no longer needed, thanks :+1: